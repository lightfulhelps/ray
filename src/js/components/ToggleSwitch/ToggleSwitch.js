/* eslint-disable jsx-a11y/label-has-associated-control */
// @flow
import * as React from 'react';
// import classNames from 'classnames';
// import Icon from '../Icon/Icon';
import type { IconNameType } from '../Icon/icons';

type Props = {
  children?: React.Node,
  className?: string,
  icon?: IconNameType,
  iconPosition?: 'left' | 'right',
  isBlock?: boolean,
  isDisabled?: boolean,
  isOutline?: boolean,
  onClick?: (SyntheticMouseEvent<>) => void,
  size?: 'lg' | 'md' | 'sm',
  tag?: string,
  theme?: string,
  type?: 'toggleSwitch' | 'submit' | 'reset',
};

const ToggleSwitch = ({ onClick, isDisabled, labelRight, labelLeft, ...other }: Props) => {
  // const classes = classNames();
  // className,

  function handleClick(e: SyntheticMouseEvent<>) {
    if (isDisabled) {
      e.preventDefault();
      return;
    }

    if (typeof onClick === 'function') onClick(e);
  }

  return (
    <div {...other} className="toggle-switch-container">
      {labelLeft && <span className="toggle-switch__label--left">{labelLeft}</span>}
      <div>
        <div className="custom-control custom-switch">
          <input
            disabled={isDisabled}
            type="checkbox"
            className="custom-control-input"
            id="customSwitch1"
          />
          <label
            className="custom-control-label"
            htmlFor="customSwitch1"
            onClick={() => handleClick()}
          />
        </div>
      </div>
      {labelRight && <span className="toggle-switch__label--left">{labelRight}</span>}
    </div>
  );
};

export default ToggleSwitch;
