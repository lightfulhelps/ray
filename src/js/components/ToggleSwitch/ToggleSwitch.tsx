/* eslint-disable jsx-a11y/label-has-associated-control */

import * as React from 'react';
import classNames from 'classnames';

type Props = {
  checked: boolean;
  className?: string;
  id?: string;
  isDisabled?: boolean;
  label?: string;
  labelAlign?: string;
  theme?: string;
  onChange?: () => void;
  onClick?: (event: React.MouseEvent) => void;
  [key: string]: any;
};

const ToggleSwitch: React.FC<Props> = ({
  className,
  checked,
  id = 'toggle-switch',
  isDisabled,
  label,
  labelAlign,
  theme,
  onChange,
  onClick,
  ...other
}: Props) => {
  let wrapperClasses = '';
  let inputClasses;
  if (labelAlign)
    switch (labelAlign) {
      case 'left':
        wrapperClasses = 'flex-row-reverse';
        inputClasses = 'ms-1';
        break;
      case 'bottom':
        wrapperClasses = 'flex-column';
        inputClasses = 'mb-1';
        break;

      case 'top':
        wrapperClasses = 'flex-column-reverse';
        inputClasses = 'mt-1';
        break;

      default:
        inputClasses = 'me-1';
        break;
    }

  const classes = classNames(className, 'form-check form-switch', wrapperClasses);

  function handleClick(e: React.MouseEvent) {
    if (isDisabled) {
      e.preventDefault();
      return;
    }

    if (typeof onClick === 'function') onClick(e);
  }

  return (
    <div {...other} className={classes}>
      <input
        type="checkbox"
        className={`form-check-input m-0${
          theme === 'primary' ? ' form-check-input-primary' : ''
        } ${inputClasses}`}
        id={id}
        disabled={isDisabled}
        role="switch"
        onChange={onChange}
        onClick={handleClick}
        checked={checked}
      />
      <label className="form-check-label" htmlFor={id}>
        {label}
      </label>
    </div>
  );
};

export default ToggleSwitch;
