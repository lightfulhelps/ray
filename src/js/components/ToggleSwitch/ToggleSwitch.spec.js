import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import ToggleSwitch from './ToggleSwitch';

const setup = (overrides = {}) => {
  const props = merge(overrides);
  const wrapper = shallow(<ToggleSwitch {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<ToggleSwitch />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
