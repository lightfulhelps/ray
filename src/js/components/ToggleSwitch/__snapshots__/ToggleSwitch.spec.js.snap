// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> should render 1`] = `
<div
  className="custom-control custom-switch"
>
  <input
    className="custom-control-input"
    id="toggle-switch"
    type="checkbox"
  />
  <label
    className="custom-control-label"
    htmlFor="toggle-switch"
    onClick={[Function]}
  />
</div>
`;

exports[`<ToggleSwitch /> should render label 1`] = `
<div
  className="custom-control custom-switch"
>
  <input
    className="custom-control-input"
    id="toggle-switch"
    type="checkbox"
  />
  <label
    className="custom-control-label"
    htmlFor="toggle-switch"
    onClick={[Function]}
  >
    this is a label
  </label>
</div>
`;
